python/ray/data/_internal/arrow_block.py

```diff
@@ -405,6 +408,47 @@ def combine(self, key: KeyFn, aggs: Tuple[AggregateFn]) -> Block[ArrowRow]:
             aggregation.
             If key is None then the k column is omitted.
         """
+        import polars as pl
+
+        if key is not None and not isinstance(key, str):
+            raise ValueError(
+                "key must be a string or None when aggregating on Arrow blocks, but "
+                f"got: {type(key)}."
+            )
+
+        return (
+            pl.from_arrow(self._table)
+            .groupby(key)
```

Comment:
If the key is sorted and polars doesn't know (because it got it from arrow).

You can add a `df.with_column(pl.col("key").set_sorted())`.

This triggers fast paths for sorted data. It's important that you only set this for sorted dat, otherwise output will be incorrect.

Reply:
Oh nice! That's great to know, I'll add that.